### PR TITLE
GH-2349: As a developer I want functions treated as pure values w.r.t hollow elements

### LINKED ISSUE #2349: As a developer I want functions treated as pure values w.r.t hollow elements

### PR DESCRIPTION
**Pull request by mmews-n4**
closes #2349

### CODE REVIEW

File: plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/ExportedElementsCollector.java [hunk 1]
```diff
@@ -158,7 +159,8 @@ private void doCollectElements(AbstractNamespace namespace, CollectionInfo info)
 	private void doCollectElement(String exportedName, TExportableElement exportedElem, CollectionInfo info) {
 
 		boolean include = (info.includeHollows || !N4JSLanguageUtils.isHollowElement(exportedElem, variantHelper))
-				&& (info.includeVariables || !(exportedElem instanceof TVariable));
+				&& (info.includeVariables || !N4JSLanguageUtils.isValueOnlyElement(exportedElem, variantHelper));
```
Comment (mor-n4) [inline plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/ExportedElementsCollector.java:162]: Would it make sense to now rename field `CollectionInfo#includeVariables` to `includeValues`? (also the corresponding parameter for method `#getExportedElements()`)

File: plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/ExportedElementsCollector.java [hunk 2]
```diff
@@ -169,7 +171,7 @@ private void doCollectElement(String exportedName, TExportableElement exportedEl
 						visibility.accessModifierSuggestion));
 			}
 		} else {
-			if (exportedElem instanceof Type) {
+			if (exportedElem instanceof Type && !(exportedElem instanceof TFunction)) {
```
Comment (mor-n4) [inline plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/ExportedElementsCollector.java:174]: Is this change necessary? (now that `TFunction` no longer inherits from `Type`)

**Review by mor-n4 — APPROVED** [review]
Great improvements!
:+1:

File: plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/ExportedElementsCollector.java (hunk 2 above)
Comment (mmews-n4) [reply plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/ExportedElementsCollector.java:174]: As discussed, I will try later to change inheritance regarding `GenericType` inheriting from `Type`.

**Comment by mmews-n4** [comment]
Thanks for your feedback!
CI was green, merging.
